<?xml version="1.0"?>

<launch>
	<include file="$(find realsense2_camera)/launch/rs_camera.launch"/>

	<arg name="base_link_id" default="base_link_uco" />
	<arg name="base_to_cam_dx" default="0.34075" />
	<arg name="base_to_cam_dy" default="0.0175" />
	<arg name="base_to_cam_dz" default="0.3" />


  	<node name="ucoslam_wrapper_node" pkg="ucoslam_wrapper" type="ucoslam_wrapper_node" output="screen" respawn="false" >
	    <param name="undistort" value="false"/>
	    <param name="skip_pose_num" value="20" />
	    <param name="map_frame_id" value="map_uco" />
	    <param name="base_link_id" value="$(arg base_link_id)" />
	    <param name="cam_uco_id" value="camera_color_optical_frame" />
	    <param name="cam_cali_file" value="$(find ucoslam_wrapper)/src/1280realsense.yml"/>
	    <param name="uco_params_file" value="$(find ucoslam_wrapper)/src/myparams.yml"/>
	    <param name="uco_map_file" value="$(find ucoslam_wrapper)/src/world.map"/>
	    <param name="skip_poor_image_cnt" value="10"/>
	    <!--param name="base_to_cam_dx" value="$(arg base_to_cam_dx)" />
	    <param name="base_to_cam_dy" value="$(arg base_to_cam_dy)" />
	    <param name="base_to_cam_dz" value="$(arg base_to_cam_dz)" /-->
  	</node>

	<node pkg="tf" type="static_transform_publisher" name="base_link_to_cam" args="$(arg base_to_cam_dx) $(arg base_to_cam_dy) $(arg base_to_cam_dz) 0 0 0 $(arg base_link_id) camera_link 100" /> 
</launch>
